R/lancelot.R

```diff
@@ -497,6 +521,40 @@ lancelot_parameters <- function(start_date, region,
                                                  vaccine_catchup_fraction,
                                                  n_doses)
 
+  ## vacc_skip parameters
+  if (length(vacc_skip_progression_rate) == 1) {
+    ret$vacc_skip_progression_rate_base <-
+      rep(vacc_skip_progression_rate, ret$n_groups)
+  } else {
+    if (length(vacc_skip_progression_rate) != ret$n_groups) {
+      stop(sprintf("vacc_skip_progression_rate must be a scalar or a vector of
+                   length %s", ret$n_groups))
+    } else {
+      ret$vacc_skip_progression_rate_base <- vacc_skip_progression_rate
+    }
+  }
+  if (vaccination$n_vacc_classes == 1) {
+    if (vacc_skip_from != 1) {
+      stop("n_vacc_classes = 1 so require vacc_skip_from = 1")
+    }
+    if (vacc_skip_to != 1) {
+      stop("n_vacc_classes = 1 so require vacc_skip_to = 1")
+    }
+  } else {
+    if (vacc_skip_to < vacc_skip_from) {
```

Comment:
should we add something else which checks that 
if (vacc_skip_to == vacc_skip_from) vacc_skip_progression_rate should be zero?

Reply:
(otherwise we do a lot of work for nothing since we move people into their own compartment...

Reply:
@edknock just bringing this up as I'm not sure it's been addressed? Also one thing I wondered about was whether we should check that vacc_skip_to >= vacc_skip_from + 2 since we want to have a check (except when n_doses = 1 maybe where we allow moves within the same compartment?). let mw know what you think

Reply:
I have revamped the checking so it is better overall, dealing with these issues